### ISSUE TITLE
Investigate Uncaught (in promise) when navigating due to time out

### ISSUE THREAD

**Issue by ankur22** [grafana/xk6-browser#970]
While running the following test script:

```js
import { check } from 'k6';
import { browser } from 'k6/experimental/browser';

export const options = {
  scenarios: {
    ui: {
      executor: 'shared-iterations',
      vus: 5,
      iterations: 1000,
      maxDuration: '40m',
      options: {
        browser: {
          type: 'chromium',
        },
      },
    },
  },
  thresholds: {
    checks: ["rate==1.0"]
  }
}

export default async function () {
  const context = browser.newContext();
  const page = context.newPage();

  try {
    // Goto front page, find login link and click it
    await page.goto('https://test.k6.io/', { waitUntil: 'networkidle' });
    await Promise.all([
      page.waitForNavigation(),
      page.locator('a[href="/my_messages.php"]').click(),
    ]);
    // Enter login credentials and login
    page.locator('input[name="login"]').type('admin');
    page.locator('input[name="password"]').type('123');
    // We expect the form submission to trigger a navigation, so to prevent a
    // race condition, setup a waiter concurrently while waiting for the click
    // to resolve.
    await Promise.all([
      page.waitForNavigation(),
      page.locator('input[type="submit"]').click(),
    ]);
    check(page, {
      'header': page.locator('h2').textContent() == 'Welcome, admin!',
    });
  } finally {
    page.close();
  }
}
```

The following errors were present in the log:

```bash
Uncaught (in promise) navigating frame to "https://test.k6.io/": navigating to "https://test.k6.io/": timed out after 30s executor=shared-iterations scenario=ui
```

The test run was for 1000 iteration, but only 998 checks were completed (all were successful though). Why were we missing 2 of the checks? It could have been caused by this time out logged (example above).

We need to investigate why this occurred and what we can do to prevent it.

Test run id: 154420

**Comment by crisreekoh** [comment]
I encounter the same issue while running browser instance script in linux machine
`ERRO[0007] Failed to load resource: the server responded with a status of 403 ()  browser_source=network line_number=0 source=browser stacktrace="<nil>" url="https://alb.reddit.com/rp.gif?ts=1689847456159&id=t2_aa7ovq9w&event=PageVisit&m.itemCount=undefined&m.value=&m.valueDecimal=undefined&m.currency=undefined&m.transactionId=&m.customEventName=&m.products=&m.conversionId=&uuid=7da1cb1a-5e29-46f8-a608-9355a4ac768b&aaid=&em=&external_id=&idfa=&integration=gtm&opt_out=0&sh=1280&sw=720&v=rdt_f5bd31b2"
ERRO[0033] Uncaught (in promise) navigating frame to "https://qa14.legalmatch.com/": navigating to "https://qa14.legalmatch.com/": timed out after 30s  executor=constant-vus scenario=browser`

**Comment by inancgumus** [comment]
Closed by #1219.
